app/controllers/ratings_controller.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+class RatingsController < ApplicationController
```

Comment:
Is this class intended to be abstract and inherited by multiple concrete "ratable" resource controllers?
If so, please use that inheritance right from the start.
If it's not planned — please avoid all that `singularize.classify.constantize` magic.

It's really very cool that you can do tricks like that — but the good practice is to only use Arcane Arts of Metaprogramming only when you really need to.